kore/src/Kore/JsonRpc.hs

```diff
@@ -734,6 +737,7 @@ respond serverState moduleName runSMT =
 
 data ServerState = ServerState
     { serializedModules :: Map.Map ModuleName SerializedDefinition
+    , receivedModules :: Map.Map ModuleName Text
```

Comment:
This is of course a very straightforward implementation to address the corner case.

I was thinking we could instead check the prior `indexedModules` for collisions: 
When we look up `newModule` in [lines 583](https://github.com/runtimeverification/haskell-backend/pull/3721/files#diff-e317d89df99d4965b340614d133c23673315aa243ae168373132bc5c9b8812f3R583-R586) we could also look it up in `indexedModules` (before the addition) as well. If we find a module `m<hash>` it must be identical to `newModule`.

However, this requires us to do part of the internalisation work  beforehand, and the comparison on the `IndexedModule`.

Reply:
I tried that and couldn't get it to work, because the internalisation code does a bunch of stuff like imports where things get muddy.... basically, I tried internalising the new module and then compare to the one that already exists, but somehow ended up with the previous module instead. Since this is a pretty unrealistic corner case, I really don't want to be spending time understanding/refactoring the internalisation to figure out where this was going wrong...